test/e2e/user_management_test.go

```diff
@@ -255,12 +298,124 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 			murMember1.Status.ProvisionedTime.String())
 
 		// The user should be set to deactivating, but not deactivated
-		_, err = s.hostAwait.WaitForUserSignup(userSignupMember1.Name, wait.UntilUserSignupHasConditions(Deactivating()...))
+		_, err = s.hostAwait.WaitForUserSignup(userSignupMember1.Name, wait.UntilUserSignupHasConditions(
+			ConditionSet(Default(), ApprovedAutomatically(), Deactivating())...))
 		require.NoError(s.T(), err)
 
 		// Verify resources have been provisioned
 		VerifyResourcesProvisionedForSignup(t, s.hostAwait, userSignupMember1, "base", s.memberAwait)
 	})
+
+	s.T().Run("test full automatic user deactivation lifecycle", func(t *testing.T) {
+		// Set configuration to 3 days
+		s.hostAwait.UpdateToolchainConfig(
+			testconfig.AutomaticApproval().Enabled(),
+			testconfig.Deactivation().DeactivatingNotificationDays(3))
+
+		hostConfig := s.hostAwait.GetToolchainConfig().Spec.Host
+		require.Equal(s.T(), 3, *hostConfig.Deactivation.DeactivatingNotificationDays)
+
+		// Create a token and identity to sign up with
+		identity0 := authsupport.NewIdentity()
+		emailValue := uuid.Must(uuid.NewV4()).String() + "@redhat.com"
+		emailClaim0 := authsupport.WithEmailClaim(emailValue)
+		token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0)
+		require.NoError(s.T(), err)
+
+		// Call the signup endpoint
+		invokeEndpoint(s.T(), "POST", s.hostAwait.RegistrationServiceURL+"/api/v1/signup", token0, "", http.StatusAccepted)
+
+		// Wait for the UserSignup to be created
+		userSignup, err := s.hostAwait.WaitForUserSignup(identity0.ID.String(),
+			wait.UntilUserSignupHasConditions(ConditionSet(Default(), ApprovedAutomatically())...),
+			wait.UntilUserSignupHasStateLabel(toolchainv1alpha1.UserSignupStateLabelValueApproved))
+		require.NoError(s.T(), err)
+
+		// Wait the Master User Record to be provisioned
+		VerifyResourcesProvisionedForSignup(s.T(), s.hostAwait, userSignup, "base", s.memberAwait)
```

Comment:
Why not using ` s.newUserRequest()` for that?

Reply:
I wrote this test before I did the other refactoring, but yes I should be using the new API.  Fixed.